domain/lib/src/usecases/received/get_all_received_shares_interactor.dart

```diff
@@ -38,18 +38,27 @@ import 'package:domain/src/repository/received/received_share_repository.dart';
 import 'package:domain/src/state/failure.dart';
 import 'package:domain/src/state/success.dart';
 import 'package:domain/src/usecases/received/received_share_view_state.dart';
+import 'package:collection/collection.dart';
+import 'package:domain/src/usecases/received/remove_deleted_received_share_from_local_database.dart';
 
 class GetAllReceivedSharesInteractor {
   final ReceivedShareRepository _receivedShareRepository;
+  final RemoveDeletedReceivedShareFromLocalDatabaseInteractor
+      _removeDeletedReceivedShareFromLocalDatabase;
 
-  GetAllReceivedSharesInteractor(this._receivedShareRepository);
+  GetAllReceivedSharesInteractor(this._receivedShareRepository,
+      this._removeDeletedReceivedShareFromLocalDatabase);
 
-  Future<Either<Failure, Success>> execute() async {
+  Future<Either<Failure, Success>> execute(String recipient) async {
     try {
       final receivedShares = await _receivedShareRepository.getAllReceivedShares()
-        .onError((error, stackTrace) => _receivedShareRepository.getAllReceivedShareOffline());
+          .onError((error, stackTrace) => _receivedShareRepository
+              .getAllReceivedShareOfflineByRecipient(recipient));
       final combinedReceivedShares = List<ReceivedShare>.empty(growable: true);
 
+      _removeDeletedReceivedShareFromLocalDatabase.execute(
+          receivedShares, recipient);
```

Comment:
We try to clean up the cache of offline mode every open Received Share screen. Am I right?

Reply:
yes, we try to clean the local database with every fetch